src/core/textrenderer/qgstextblockformat.cpp

```diff
@@ -55,6 +57,31 @@ void QgsTextBlockFormat::overrideWith( const QgsTextBlockFormat &other )
     mHorizontalAlign = other.mHorizontalAlign;
     mHasHorizontalAlignSet = true;
   }
+
+  if ( std::isnan( mLineHeight ) )
+    mLineHeight = other.mLineHeight;
+  if ( std::isnan( mLineHeightPercentage ) )
+    mLineHeightPercentage = other.mLineHeightPercentage;
```

Comment:
Why are these values only copied when the other value is nan?

Reply:
Note to self, answer is in the .h[eader] : https://github.com/qgis/QGIS/blob/3190c5faee8963a43e353f101480bee0e13043ed/src/core/textrenderer/qgstextblockformat.h#L61-L70

Reply:
The method is a little counter-intuitive (but it's all documented in the docstrings in the header), but it is designed to replace any not-set properties with values inherited from a parent containing block